scanner/src/main/kotlin/experimental/NestedProvenanceScanResult.kt

```diff
@@ -136,4 +137,70 @@ data class NestedProvenanceScanResult(
 
         throw IllegalArgumentException("Could not find entry for $provenance.")
     }
+
+    /**
+     * Remove all scan results for [nestedProvenance]s which are not within the provided [path] and filter all findings
+     * within the scan results by [path], taking the sub repository paths into account.
+     * This also updates the VCS paths for all [RepositoryProvenance]s accordingly to make clear that they are not
+     * results for the whole repository.
+     *
+     * Throws an [IllegalArgumentException] if any provenance in [nestedProvenance] already has a VCS path set.
+     */
+    fun filterByVcsPath(path: String): NestedProvenanceScanResult {
+        if (path.isBlank()) return this
+
+        require(getProvenances().all { it !is RepositoryProvenance || it.vcsInfo.path.isBlank() }) {
+            "filterByVcsPath() requires that all provenances have no VCS path set."
+        }
+
+        val pathsWithinProvenances = getProvenances().filter {
+            val provenancePath = getPath(it)
+            provenancePath.isBlank() || provenancePath.startsWith("$path/")
```

Comment:
Can you please add a code comment why you allow blank paths here? (I guess because root license files need to be considered in any case.)

Reply:
The code was actually wrong, with the fixed code it's hopefully more obvious that the blank path is treated separately, because paths of findings are not prefixed with "/":
```provenancePath.isBlank() || path.startsWith("$provenancePath/")```